src/Cli/dotnet/commands/dotnet-sln/add/file/AddFileToSolutionCommand.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Sln.Internal;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.Common;
+
+namespace Microsoft.DotNet.Tools.Sln.Add;
+
+internal class AddFileToSolutionCommand : CommandBase
+{
+    private readonly string _fileOrDirectory;
+    private readonly bool _inRoot;
+    private readonly IList<string> _relativeRootSolutionFolders;
+    private readonly IReadOnlyList<string> _arguments;
+
+    public AddFileToSolutionCommand(ParseResult parseResult) : base(parseResult)
+    {
+        var projectPaths = parseResult.GetValue(SlnAddParser.ProjectPathArgument)?.ToArray() ?? [];
+        if (projectPaths.Length != 0)
+        {
+            throw new GracefulException(LocalizableStrings.ProjectPathArgumentShouldNotBeProvidedForDotnetSlnAddFile);
+        }
+
+        _fileOrDirectory = parseResult.GetValue(SlnCommandParser.SlnArgument);
+        _arguments = parseResult.GetValue(SlnAddFileParser.FilePathArgument).ToArray();
+        _inRoot = parseResult.GetValue(SlnAddFileParser.InRootOption);
+        string relativeRoot = parseResult.GetValue(SlnAddFileParser.SolutionFolderOption);
+
+        SlnArgumentValidator.ParseAndValidateArguments(_arguments, SlnArgumentValidator.CommandType.Add, _inRoot, relativeRoot, subcommand: "file");
+
+        if (string.IsNullOrEmpty(relativeRoot))
+        {
+            _relativeRootSolutionFolders = null;
+        }
+        else
+        {
+            relativeRoot = PathUtility.GetPathWithDirectorySeparator(relativeRoot);
+            _relativeRootSolutionFolders = relativeRoot.Split(Path.DirectorySeparatorChar, StringSplitOptions.RemoveEmptyEntries);
+        }
+    }
+
+    public override int Execute()
+    {
+        SlnFile slnFile = SlnFileFactory.CreateFromFileOrDirectory(_fileOrDirectory);
+
+        var arguments = _parseResult.GetValue(SlnAddFileParser.FilePathArgument).ToArray();
+        if (arguments.Length == 0)
+        {
+            throw new GracefulException(LocalizableStrings.SpecifyAtLeastOneFileToAdd);
+        }
+
+        PathUtility.EnsureAllPathsExist(arguments, LocalizableStrings.CouldNotFindFile, allowDirectories: false);
+
+        var relativeFilePaths = _arguments.Select(f =>
+        {
+            var fullFilePath = Path.GetFullPath(f);
+            return Path.GetRelativePath(
+                PathUtility.EnsureTrailingSlash(slnFile.BaseDirectory),
+                fullFilePath);
+        }).ToList();
+
+        // Perform the same validations as Visual Studio:
```

Comment:
![image](https://github.com/user-attachments/assets/2e7574c8-5282-4ea1-9db6-1aa7372ff0c7)
Didn't check for user options files; should I? If so, what files should I be looking for?